### PR TITLE
Implement Array#pack c- and C-directives

### PR DESCRIPTION
**Pull request by richardboehme**
This also adds an `Encoding::BINARY` constant as it's needed for one of the specs.

### CODE REVIEW

**Comment by richardboehme** [comment]
This currently seems to be incomplete (even though the specs already pass)...

**Comment by seven1m** [comment]
@richardboehme what do you mean? Should we merge this as-is? Or are you saying there's a problem?

**Comment by richardboehme** [comment]
> @richardboehme what do you mean? Should we merge this as-is? Or are you saying there's a problem?

There's definitely some misbehavior regarding larger numbers (> 256) because the c-directive should print 8 bit integers. Also I guess there must be some difference between c and C even though the spec treats it in the same way... So I'd say to not merge this as-is. I want to correct the behavior first.

**Comment by richardboehme** [comment]
> There's definitely some misbehavior regarding larger numbers (> 256) because the c-directive should print 8 bit integers. Also I guess there must be some difference between c and C even though the spec treats it in the same way... So I'd say to not merge this as-is. I want to correct the behavior first.

First point seemed to be an illusion.... String#inspect yielded different results for some larger numbers because we only print them in the format '\xXX' if the character is less than 32. This resulted in Natalie giving me "�" when MRI returned "\x02". But they are the same so this should be fine!

I tried finding a difference between the 'c'- and the 'C'-directive but despite the difference of their definition (signed vs unsigned char) there does not seem to be any difference in calculating the result for those? I'm just giving up on this until somebody finds a bug with it 😅 

And lastly I added support for Bignums to our IntegerPacker. I hope making m_source an IntegerValue* instead of a nat_int_t is fine for you?

From my point of view we could merge this now 😄

**Comment by seven1m** [comment]
Great work!

File: test/natalie/array_test.rb [hunk 1]
```diff
@@ -1180,6 +1180,17 @@ def x.to_a
     end
   end
 
+  describe '#pack' do
+    describe 'c' do
+      fit 'handle bignums' do
```
Comment (seven1m) [inline test/natalie/array_test.rb:1185]: Just noticed this is focused @richardboehme :grimacing:

Comment (richardboehme) [reply test/natalie/array_test.rb:1185]: Fixed in #266 😅

**Pull request merged**